code/controllers/subsystem/lobotomy_corp.dm

```diff
@@ -101,13 +104,21 @@ SUBSYSTEM_DEF(lobotomy_corp)
 	return TRUE
 
 // Called when any normal midnight ends
-/datum/controller/subsystem/lobotomy_corp/proc/PickPotentialSuppressions()
+/datum/controller/subsystem/lobotomy_corp/proc/PickPotentialSuppressions(announce = FALSE, extra_core = FALSE)
 	if(istype(core_suppression))
 		return
 	if(!LAZYLEN(GLOB.abnormality_auxiliary_consoles)) // There's no consoles, for some reason
 		message_admins("Tried to pick potential core suppressions, but there was no auxiliary consoles! Fix it!")
 		return
 	var/list/cores = subtypesof(/datum/suppression)
+	// Remove cores that don't fit requirements
+	for(var/core_type in cores)
+		var/datum/suppression/C = core_type
+		if(!extra_core && initial(C.after_midnight))
+			cores -= core_type
+			continue
+		if(extra_core && !initial(C.after_midnight))
+			cores -= core_type
```

Comment:
```suggestion
	for(var/datum/suppression/core_type in cores)
		if(!extra_core && initial(core_type.after_midnight))
			cores -= core_type
			continue
		if(extra_core && !initial(core_type.after_midnight))
			cores -= core_type
```
Why not this way?

Reply:
Because this will not work, how comes you don't know that?
`for(var/type/T in list)` will loop through all objects of this **type**, ignoring others. Know what that means? It will not consider paths, which is exactly what `cores` list has.